devtools/gossipwith.c

```diff
@@ -137,11 +156,21 @@ static struct io_plan *handshake_success(struct io_conn *conn,
 				OPTIONAL_FEATURE(OPT_INITIAL_ROUTING_SYNC));
 
 	if (!no_init) {
-		msg = towire_init(NULL, NULL, features, NULL);
+		struct tlv_init_tlvs *tlvs = NULL;
+		if (blockchainparams) {
+			tlvs = tlv_init_tlvs_new(NULL);
+			tlvs->networks = tal(tlvs, struct tlv_init_tlvs_networks);
+			tlvs->networks->chains = tal_arr(tlvs->networks, struct bitcoin_blkid, 1);
+			tlvs->networks->chains[0] = blockchainparams->genesis_blockhash;
+			msg = towire_init(NULL, NULL, features, tlvs);
+		} else
+			msg = towire_init(NULL, NULL, features, NULL);
 
 		sync_crypto_write(pps, take(msg));
 		/* Ignore their init message. */
 		tal_free(sync_crypto_read(NULL, pps));
+		if (tlvs)
+			tal_free(tlvs);
```

Comment:
Might be an option to make `towire_init` `take()` the `tlvs` and therefore hand off ownership.

Reply:
Yes, but should be an other PR I think ?

Reply:
And a big one!  There are a few places we may hand pointers *assuming* generated routines don't take()...